core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/function/aggregate/var/VarianceCollector.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eclipse RDF4J contributors.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *******************************************************************************/
+package org.eclipse.rdf4j.query.algebra.evaluation.function.aggregate.var;
+
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.base.CoreDatatype;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.eclipse.rdf4j.query.algebra.MathExpr;
+import org.eclipse.rdf4j.query.algebra.evaluation.function.aggregate.StatisticCollector;
+import org.eclipse.rdf4j.query.algebra.evaluation.util.MathUtil;
+
+/**
+ * {@link org.eclipse.rdf4j.query.parser.sparql.aggregate.AggregateCollector} that can compute both sample and
+ * population variance based on input of numeric {@link Literal}s
+ *
+ * @author Tomas Kovachev t.kovachev1996@gmail.com
+ */
+public class VarianceCollector extends StatisticCollector {
+	private static final Literal ONE = SimpleValueFactory.getInstance().createLiteral("1", CoreDatatype.XSD.INTEGER);
+
+	protected Literal sum;
+
+	public VarianceCollector(boolean population) {
+		super(population);
+		this.sum = SimpleValueFactory.getInstance().createLiteral("0", CoreDatatype.XSD.INTEGER);
+	}
+
+	@Override
+	public void addValue(Literal val) {
+		sum = MathUtil.compute(sum, val, MathExpr.MathOp.PLUS);
+		super.addValue(val);
+	}
+
+	protected Literal computeValue() {
+		var size = SimpleValueFactory.getInstance().createLiteral(values.size());
+		var mean = MathUtil.compute(sum, size, MathExpr.MathOp.DIVIDE);
+		var sqDifferenceSum = SimpleValueFactory.getInstance().createLiteral("0", CoreDatatype.XSD.INTEGER);
```

Comment:
Could you use a static constant literal `ZERO` here?

Reply:
Sure, just updated it.